packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolAncestors.test.ts

```diff
@@ -0,0 +1,100 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { Testing } from '@defichain/jellyfish-testing'
+import BigNumber from 'bignumber.js'
+
+describe('transactions without ancestors', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should return empty array for transaction id without ancestors', async () => {
+    const txId = await testing.rpc.wallet.sendToAddress('mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU', 0.003)
+    const mempoolEntry = await testing.rpc.blockchain.getMempoolAncestors(txId)
+
+    expect(mempoolEntry.length).toStrictEqual(0)
+  })
+
+  it('should return error if transaction is not in mempool', async () => {
+    const txId = '9fb9c46b1d12dae8a4a35558f7ef4b047df3b444b1ead61d334e4f187f5f58b7'
+    await expect(testing.rpc.blockchain.getMempoolAncestors(txId))
+      .rejects
+      .toThrowError('RpcApiError: \'Transaction not in mempool\', code: -5, method: getmempoolancestors')
+  })
+
+  it('should return error if transaction id has an invalid length', async () => {
+    await expect(testing.rpc.blockchain.getMempoolAncestors('invalidtxidstring'))
+      .rejects
+      .toThrow('RpcApiError: \'parameter 1 must be of length 64 (not 17, for \'invalidtxidstring\')\', code: -8, method: getmempoolancestors')
+  })
+})
+
+describe('transactions with ancestors', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    for (let i = 0; i < 3; i++) {
+      await testing.rpc.wallet.sendToAddress('mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU', 0.003)
+    }
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should return JSON object if verbose is true', async () => {
+    const txId = await testing.rpc.wallet.sendToAddress('mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU', 0.003)
+    const mempoolAncestors = await testing.rpc.blockchain.getMempoolAncestors(txId, true)
+    const keys = Object.keys(mempoolAncestors)
+    expect(keys.length).toBeGreaterThan(0)
+    for (const key of keys) {
```

Comment:
Would prefer to have some stricter check here (e.g. asserting the exact count of the mempool ancestors for the given tx)

Reply:
see my comment above, it would be tricky to assert the exact count of mempool ancestors for a particular transaction (unless there is only one other transaction in the mempool)